server/mqtt.go

```diff
@@ -2553,7 +2536,8 @@ func (as *mqttAccountSessionManager) transferRetainedToPerKeySubjectStream(log *
 		}
 		// Store the message again, this time with the new per-key subject.
 		subject := mqttRetainedMsgsStreamSubject + rmsg.Subject
-		if _, err := jsa.storeMsgWithKind(mqttJSASessPersist, subject, 0, smsg.Data); err != nil {
+
```

Comment:
This was likely a mistake. I think using `jsa.storeMsgWithKind(mqttJSASessPersist,...` here was causing unnecessary JS ACK processing. @neilalexander Please confirm sice it came from https://github.com/nats-io/nats-server/pull/4199

Reply:
Not sure I'm following, this is only for the migration from the old subject? If the migration has completed already then I wouldn't expect us to get this far.

Reply:
Correct, I think this is unrelated to the customer issue; just checking that the intention is to just save the message, not the `SP` ack broadcast. Like you said, it's a one-time migration.

Reply:
Good catch. I think that this was an issue when implementing the move of retained messages with code copied from the move of session records.